clients/admin-ui/src/features/common/custom-fields/CreateCustomFields.tsx

```diff
@@ -87,43 +67,45 @@ const CreateCustomFields = forwardRef(
       onSubmitComplete();
     };
 
-    const loadList = useCallback((): Map<string, ItemOption> => {
-      const list = new Map<string, ItemOption>();
-      if (data?.length) {
-        data.forEach((value, index) => {
-          if (value?.id && value?.name) {
-            list.set(value.name, { value: value.id });
-            if (index === 0) {
-              initialValues.allow_list_id = value.id;
```

Comment:
[Post-review]: Mutating object like this is _very_ error prone. Not only was this a dirty way of populating the object on first render, but this was also modifying a **module scoped constant**. So this value could stick around even when the component was rendered for a completely different field.